From adf9ce70ae7399a5cd4f1878e8c782f00729e936 Mon Sep 17 00:00:00 2001
From: Vishal Sagar <vishal.sagar@xilinx.com>
Date: Fri, 28 Sep 2018 11:05:07 +0530
Subject: [PATCH 0432/1851] drm: xlnx: scaler: Fix array out of bound access

commit 011ad5ee79eb350cb4b448ef27e83c00e64e96c3 from
https://github.com/Xilinx/linux-xlnx.git

This patch fixes random kernel crashes observed due to accessing and
modifying the out of bound address of the vertical scaler coefficients
array.

Signed-off-by: Vishal Sagar <vishal.sagar@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/gpu/drm/xlnx/xlnx_scaler.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/drivers/gpu/drm/xlnx/xlnx_scaler.c b/drivers/gpu/drm/xlnx/xlnx_scaler.c
index d96ddeb49b59..0c13d3cd5728 100644
--- a/drivers/gpu/drm/xlnx/xlnx_scaler.c
+++ b/drivers/gpu/drm/xlnx/xlnx_scaler.c
@@ -1059,10 +1059,11 @@ xv_vscaler_load_ext_coeff(struct xilinx_scaler *scaler,
 			/* pad left */
 			for (j = 0; j < offset; j++)
 				scaler->vscaler_coeff[i][j] = 0;
+			/* pad right */
+			j = ntaps + offset;
+			for (; j < XV_VSCALER_MAX_V_TAPS; j++)
+				scaler->vscaler_coeff[i][j] = 0;
 		}
-		/* pad right */
-		for (j = (ntaps + offset); j < XV_VSCALER_MAX_V_TAPS; j++)
-			scaler->vscaler_coeff[i][j] = 0;
 	}
 }
 
-- 
2.31.1

